tfhe/src/shortint/public_key/compact.rs

```diff
@@ -150,8 +155,8 @@ impl CompactPublicKey {
                     &self.key,
                     &mut ct_list,
                     &plaintext_list,
-                    self.parameters.glwe_modular_std_dev(),
-                    self.parameters.lwe_modular_std_dev(),
+                    encryption_noise,
```

Comment:
Not sure how it works, but is it expected to have `mask_noise_parameters = body_noise_parameters`?
Then why have separate function arguments?

Reply:
Marc's scheme is generic on that but we use the same distribution

Reply:
you can check, we have the same elsewhere, the idea to reason about it is that the noise is done for the body of size N of a GLWE for example or of size n for an LWE, we have a mask and a body of the same size in the public key, so using the noise for the GLWE in the Big Key case and the LWE noise in the small key case means we are secure in the encryption as we are adding noise in polynomials which have the same size